sdk/go/pulumi-language-go/main.go

```diff
@@ -358,20 +362,48 @@ func (m *modInfo) getPlugin(moduleRoot string) (*pulumirpc.PluginDependency, err
 	return plugin, nil
 }
 
-// Reads and parses the go.mod file from the current working directory.
-func (host *goLanguageHost) loadGomod(moduleDir string) (*modfile.File, error) {
-	path := filepath.Join(moduleDir, "go.mod")
-	body, err := os.ReadFile(path)
+// Reads and parses the go.mod file for the program at the given path.
+// Returns the parsed go.mod file and the path to the module directory.
+// Relies on the 'go' command to find the go.mod file for this program.
+func (host *goLanguageHost) loadGomod(gobin, programDir string) (modDir string, modFile *modfile.File, err error) {
+	// Get the path to the go.mod file.
+	// This may be different from the programDir if the Pulumi program
+	// is in a subdirectory of the Go module.
+	//
+	// The '-f {{.GoMod}}' specifies that the command should print
+	// just the path to the go.mod file.
+	//
+	//	type Module struct {
+	//		Path     string // module path
+	//		...
+	//		GoMod    string // path to go.mod file
+	//	}
+	//
+	// See 'go help list' for the full definition.
```

Comment:
The comments and docstring for this function are really great! <3

I would just add a sentence to say that if you're in a folder that doesn't have a `go.mod` file in it, or above it, then the expected output from this exec command is the empty string (and not, for example, a non-zero exit code).

Reply:
Done